config/gulp.js

```diff
@@ -1,103 +1,117 @@
-'use strict'
+'use strict';
 
-const gulp = require('gulp')
-const babel = require('gulp-babel')
-const sass = require('gulp-sass')
-const webpack = require('webpack-stream')
-const watch = require('gulp-watch')
-const path = require('path')
+const gulp = require('gulp');
+const babel = require('gulp-babel');
+const sass = require('gulp-sass');
+const webpack = require('webpack-stream');
+const path = require('path');
 
 const babelConfig = {
-  presets: ['react', 'es2015'],
-  plugins: [
-    [
-      'css-modules-transform', {
-        preprocessCss: 'preprocessor.js',
-        generateScopedName: '[name]__[local]___[hash:base64:5]',
-        extensions: ['.css', '.scss'],
-        extractCss: './dist/styles/production.css'
-      }
+    presets: ['react', 'es2015'],
+    plugins: [
+        [
+            'css-modules-transform', {
+                preprocessCss: 'preprocessor.js',
+                generateScopedName: '[name]__[local]___[hash:base64:5]',
+                extensions: ['.css', '.scss'],
+                extractCss: './dist/styles/production.css'
+            }
+        ]
     ]
-  ]
-}
+};
 
 module.exports = {
 
-  defaultTaskName: 'default',
+    defaultTaskName: 'default',
 
-  tasks: {
+    tasks: {
 
-    default: ['compileTemplate', 'compileStyles', 'moveImages', 'bundleAdmin', 'bundleClient', 'watch'],
+        default: ['compileTemplate', 'compileStyles', 'moveImages', 'bundleAdmin', 'bundleClient', 'watchTemplates', 'watchStyles', 'watchImages'],
 
-    watch: () => {
-      return gulp.watch(
-        path.join('frontend', '**', '*'),
-        ['compileTemplate', 'compileStyles', 'moveImages', 'bundleAdmin', 'bundleClient'])
-    },
+        watchTemplates: () => {
+            return gulp.watch(
+                path.join('frontend', 'js', '**', '*'),
+                ['compileTemplate', 'bundleAdmin', 'bundleClient']
```

Comment:
sounds ok. Since it produced 3 separate bundles wasn't sure if there was an advantage in splitting up the watchers also so all 3 bundles weren't rebuilt. 

Testing it locally was much faster regardless